RecoEgamma/PhotonIdentification/plugins/PhotonXGBoostProducer.cc

```diff
@@ -0,0 +1,142 @@
+      tokenE2x2_(consumes<reco::RecoEcalCandidateIsolationMap>(config.getParameter<edm::InputTag>("inputTagE2x2"))),
+      tokenIso_(consumes<reco::RecoEcalCandidateIsolationMap>(config.getParameter<edm::InputTag>("inputTagIso"))),
+      mvaFileXgbB_(config.getParameter<edm::FileInPath>("mvaFileXgbB")),
+      mvaFileXgbE_(config.getParameter<edm::FileInPath>("mvaFileXgbE")),
+      mvaNTreeLimitB_(config.getParameter<unsigned int>("mvaNTreeLimitB")),
+      mvaNTreeLimitE_(config.getParameter<unsigned int>("mvaNTreeLimitE")),
+      mvaThresholdEt_(config.getParameter<double>("mvaThresholdEt")) {
+  mvaEstimatorB_ = std::make_unique<PhotonXGBoostEstimator>(mvaFileXgbB_, mvaNTreeLimitB_);
+  mvaEstimatorE_ = std::make_unique<PhotonXGBoostEstimator>(mvaFileXgbE_, mvaNTreeLimitE_);
+  produces<reco::RecoEcalCandidateIsolationMap>();
+}
+
+void PhotonXGBoostProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("candTag", edm::InputTag("hltEgammaCandidatesUnseeded"));
+  desc.add<edm::InputTag>("inputTagR9", edm::InputTag("hltEgammaR9IDUnseeded", "r95x5"));
+  desc.add<edm::InputTag>("inputTagHoE", edm::InputTag("hltEgammaHoverEUnseeded"));
+  desc.add<edm::InputTag>("inputTagSigmaiEtaiEta",
+                          edm::InputTag("hltEgammaClusterShapeUnseeded", "sigmaIEtaIEta5x5NoiseCleaned"));
+  desc.add<edm::InputTag>("inputTagE2x2", edm::InputTag("hltEgammaClusterShapeUnseeded", "e2x2"));
+  desc.add<edm::InputTag>("inputTagIso", edm::InputTag("hltEgammaEcalPFClusterIsoUnseeded"));
+  desc.add<edm::FileInPath>("mvaFileXgbB",
+                            edm::FileInPath("RecoEgamma/PhotonIdentification/data/xgb_photonmva_barrel_v1.bin"));
+  desc.add<edm::FileInPath>("mvaFileXgbE",
+                            edm::FileInPath("RecoEgamma/PhotonIdentification/data/xgb_photonmva_endcap_v1.bin"));
+  desc.add<unsigned int>("mvaNTreeLimitB", 55);
+  desc.add<unsigned int>("mvaNTreeLimitE", 48);
+  desc.add<double>("mvaThresholdEt", 0);
+  descriptions.addWithDefaultLabel(desc);
+}
+
+void PhotonXGBoostProducer::produce(edm::StreamID, edm::Event& event, edm::EventSetup const& setup) const {
+  edm::Handle<reco::RecoEcalCandidateCollection> recCollection;
+  event.getByToken(candToken_, recCollection);
+
+  //get hold of r9 association map
+  edm::Handle<reco::RecoEcalCandidateIsolationMap> r9Map;
+  event.getByToken(tokenR9_, r9Map);
+
+  //get hold of HoE association map
+  edm::Handle<reco::RecoEcalCandidateIsolationMap> hoEMap;
+  event.getByToken(tokenHoE_, hoEMap);
+
+  //get hold of isolated association map
+  edm::Handle<reco::RecoEcalCandidateIsolationMap> sigmaiEtaiEtaMap;
+  event.getByToken(tokenSigmaiEtaiEta_, sigmaiEtaiEtaMap);
+
+  //get hold of e2x2 (s4) association map
+  edm::Handle<reco::RecoEcalCandidateIsolationMap> e2x2Map;
+  event.getByToken(tokenE2x2_, e2x2Map);
```

Comment:
```suggestion
  const auto& recCollection = event.getHandle(candToken_);

  //get hold of r9 association map
  const auto& r9Map = event.getHandle(tokenR9_);

  //get hold of HoE association map
  const auto& hoEMap = event.getHandle(tokenHoE_);

  //get hold of isolated association map
  const auto& sigmaiEtaiEtaMap = event.getHandle(tokenSigmaiEtaiEta_);

  //get hold of e2x2 (s4) association map
  const auto& e2x2Map = event.getHandle(tokenE2x2_);
```

I am not going to insist on this though.

Reply:
sorry, forgot this ones..